frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -432,6 +433,38 @@ function getStackedDataLabelTransform(
   };
 }
 
+function getBarSeriesDataLabelTransform(
+  settings: ComputedVisualizationSettings,
+  seriesModels: SeriesModel[],
+): ConditionalTransform {
+  const barSeriesModels = seriesModels.filter(seriesModel => {
+    const seriesSettings = settings.series(
+      seriesModel.legacySeriesSettingsObjectKey,
+    );
+    return seriesSettings.display === "bar";
+  });
+
+  return {
+    condition: barSeriesModels.length > 0,
+    fn: (datum: Datum) => {
+      const transforedDatum = { ...datum };
+
+      barSeriesModels.forEach(({ dataKey }) => {
```

Comment:
With this approach it does not show values for zeros so they are indistinguishable from nulls. Wdyt about not splitting values labels to negative and positive and use a scatter series for data labels? It is used on the waterfall charts

Reply:
Good point, thanks for noticing this! Zeroes being indistinguishable from nulls seems to have always been an issue for stacked bar labels, so it's technically a separate issue (although it's effects are propagated here).

<img width="1296" alt="Screenshot 2024-05-31 at 9 23 50 AM" src="https://github.com/metabase/metabase/assets/37751258/1751c076-bf2b-492b-ba4b-4da9a630b8eb">

Since this PR aims to use the same code as stacked bar labels, I think we should open a separate issue, and refactor both stacked bar labels and these bars to use scatter series in a follow up PR. It will keep the PR size smaller, separate concerns, and allow this to land sooner (I've already spent a long time rebasing this as master keeps changing). How does that sound?

Reply:
The issue I mentioned is not related to stacked bars, it exists on non-stacked bar charts where using stacking for labels is an implementation detail that leads to extra complexity in the code. Using a single scatter series for data labels looks like an [easy and very quick change](https://github.com/metabase/metabase/pull/41296/files#diff-0224072d7446b36947d3d39e5c87804b6cde50a61598d6d906e1cbc11ca4ac9fR285) that mostly removes code so I would address it here right away which would also fix the issue with zero labels

Reply:
My judgement was wrong here, positioning scatter labels within non-stacked bars group is PITA.
Your approach is more reasonable. I created a PR that fixes zero labels: https://github.com/metabase/metabase/pull/43484